### PR TITLE
Fixing sticky header crash.

### PR DESCRIPTION
**Pull request by MattFoley**

### CODE REVIEW

File: React/Views/RCTScrollView.m [hunk 1]
```diff
@@ -345,6 +345,12 @@ - (UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event
   __block UIView *hitView;
 
   [_stickyHeaderIndices enumerateIndexesWithOptions:0 usingBlock:^(NSUInteger idx, BOOL *stop) {
+    NSUInteger subviewCount = [self contentView].reactSubviews.count;
```
Comment (greghe) [inline React/Views/RCTScrollView.m]: Can the subview count change during iteration? if not you might want to hoist this declaration outside of the block.

File: React/Views/RCTScrollView.m [hunk 2]
```diff
@@ -345,6 +345,12 @@ - (UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event
...
+    if (idx >= subviewCount) {
+      RCTLogWarn(@"Sticky header index %zd was outside the range {0, %zd}", idx, subviewCount);
+      return;
```
Comment (greghe) [inline React/Views/RCTScrollView.m]: Seems to me that if this index is out of range then any later ones will be as well. Might as well halt the enumeration: `*stop = YES;`

File: React/Views/RCTScrollView.m (hunk 1 above)
Comment (MattFoley) [inline React/Views/RCTScrollView.m]: I think it can, I think that might be what's causing the problem?

File: React/Views/RCTScrollView.m (hunk 2 above)
Comment (MattFoley) [inline React/Views/RCTScrollView.m]: Good point, will add that.

**Pull request merged**